phases/14-agent-engineering/35-initialization-scripts/code/main.py

```
@@ -74,12 +104,109 @@ def probe_state_freshness() -> Probe:
+
+def main(argv: list[str] | None = None) -> int:
+    ap = argparse.ArgumentParser()
+    ap.add_argument("--no-cache", action="store_true", help="ignore prereqs.lock and run every probe")
+    ap.add_argument("--write-lkg", action="store_true", help="pin current HEAD as last-known-good")
+    args = ap.parse_args(argv)
+
     WORK.mkdir(exist_ok=True)
+
+    if args.write_lkg:
+        try:
+            head = subprocess.check_output(["git", "rev-parse", "HEAD"], cwd=HERE, text=True).strip()
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)